pkg/metricfamily/label.go

```diff
@@ -9,6 +13,7 @@ type LabelRetriever interface {
 type label struct {
 	labels    map[string]*clientmodel.LabelPair
 	retriever LabelRetriever
+	mu        *sync.Mutex
```

Comment:
can you leave a small comment why this must be a pointer?

Reply:
it doesn't _have_ to be

Reply:
Ah, I was assuming it must be because of usages of the `label` struct as values rather than as references. In that case, actually a nit, I suggest to use a value instead of a pointer, it seems more idiomatic. But leaving the final call to you :-)

Reply:
Yeah, that was actually my first reaction as well, but then saw that the label struct is actually only referenced by reference and not value. I'll update it